omnigibson/systems/macro_particle_system.py

```diff
@@ -884,11 +987,14 @@ def _serialize(cls, state):
         state_group_flat = [[state["n_groups"]]]
         for group_name, group_dict in groups_dict.items():
             group_obj_link2id = {link_name: i for i, link_name in enumerate(cls._group_objects[group_name].links.keys())}
+            # # Assign -1 if there's no link
+            # group_obj_link2id[None] = -1
             state_group_flat += [
                 [group_dict["particle_attached_obj_uuid"]],
                 [group_dict["n_particles"]],
                 group_dict["particle_idns"],
-                [group_obj_link2id[link_name] for link_name in group_dict["particle_attached_link_names"]],
+                [group_obj_link2id[link_name] if isinstance(link_name, str) else link_name
```

Comment:
hmmm this looks super hacky, so `link_name` is not actually link name?

Reply:
fixed